astropy/time/core.py

```diff
@@ -1905,6 +1911,22 @@ def __gt__(self, other):
     def __ge__(self, other):
         return self._time_comparison(other, operator.ge)
 
+    def __array_ufunc__(self, ufunc, method, *args, **kwargs):
+        """Override numpy ufuncs. For now only isnan and isnat are implemented."""
+        if ufunc in (np.isnan, np.isnat):
+            return self._isnat()
+
+        if ufunc is np.isfinite:
+            return ~self._isnat()
+
+        return NotImplemented
+
+    def _isnat(self):
+        if self.masked:
+            return np.isnan(self.jd2.filled(np.nan))
```

Comment:
Do we really want to treat every masked value as `NaT`? The concept so far has been that masks just mask, not invalidate.

Reply:
I guess the correct thing would be to return a masked boolean array, right? Where values of the mask are also masked.

Reply:
btw, if `Time` should behave like a masked array, why doesn't it accept a `mask` kwarg?

Reply:
Edit: moved from above, this belonged here:

But I see a lot of interactions of `mask` and the check if `jd2` is nan. So what's the actual design of the internal state supposed to be?

How do we differentiated "masked" from "has a nan value" in the internal state?

Reply:
> I guess the correct thing would be to return a masked boolean array, right? Where values of the mask are also masked.

Yes, I think that would be the correct model!

> But I see a lot of interactions of `mask` and the check if `jd2` is nan.

Hmm, that is actually somewhat weird, and perhaps they are leftovers from when the mask itself was carried by `nan` -- did I miss them when I converted things to use the `Masked` class? Anyway, better discussed when I have a bit more time, with concrete examples.